spec/controllers/api/v1/base_controller_spec.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Api::V1::BaseController, type: :controller do
+  controller do
+    before_action :require_authentication!
+
+    def index
+      render plain: "index called"
+    end
+  end
+
+  describe "#index" do
+    context "when invalid user id given" do
+      it "returns a 401 unauthorized" do
+        user = FactoryBot.create(:user)
+        request.headers["X-User-Id"] = "nope"
+        request.headers["Authorization"] = "Token #{user.authentication_token}"
+
+        get :index
+
+        expect(response.status).to eq(401)
+      end
+    end
+
+    context "when no valid user id given" do
+      it "returns a 401 unauthorized" do
+        user = FactoryBot.create(:user)
+        request.headers["Authorization"] = "Token #{user.authentication_token}"
+
+        get :index
+
+        expect(response.status).to eq(401)
+      end
+    end
+
+    context "when invalid token given" do
+      it "returns a 401 unauthorized" do
+        user = FactoryBot.create(:user)
+        request.headers["X-User-Id"] = user.id
+        request.headers["Authorization"] = "Token 'nope'"
+
+        get :index
+
+        expect(response.status).to eq(401)
+      end
+    end
+
+    context "when no valid token given" do
+      it "returns a 401 unauthorized" do
+        user = FactoryBot.create(:user)
+        request.headers["X-User-Id"] = user.id
+
+        get :index
+
+        expect(response.status).to eq(401)
+      end
+    end
+
+    context "when invalid token and user id given" do
+      it "returns a 401 unauthorized" do
+        request.headers["X-User-Id"] = "nope"
+        request.headers["Authorization"] = "Token 'nope'"
+
+        get :index
+
+        expect(response.status).to eq(401)
+      end
+    end
+
+    context "when no valid token and user id given" do
+      it "returns a 401 unauthorized" do
+        get :index
+
+        expect(response.status).to eq(401)
```

Comment:
Can we do this: 
`expect(response).to have_http_status(:unathorized)`

Reply:
Is it an improvement?

Reply:
does make it more readable, arguably the numbers are for computers and the words for humans 💃

Reply:
I'm not sure it is an improvement, since it adds a bunch of syntactic sugar over what was previously a straightforward assertion. And now you need to remember that 401 is `:unauthorized` when you're comparing numeric values returned by your test failures.

But it is more rspec-like